storage/src/test/java/tech/pegasys/artemis/storage/client/RecentChainDataTest.java

```diff
@@ -189,7 +189,33 @@ public void getStateBySlot_returnGenesisBlockWhenItIsTheBestState() {
     when(store.getBlockState(GENESIS_BLOCK_ROOT)).thenReturn(newState);
     when(store.getBlock(GENESIS_BLOCK_ROOT)).thenReturn(GENESIS_BLOCK);
 
-    assertThat(storageClient.getBlockBySlot(UnsignedLong.ZERO)).contains(GENESIS_BLOCK);
+    assertThat(storageClient.getStateInEffectAtSlot(UnsignedLong.ZERO)).contains(newState);
+  }
+
+  @Test
+  public void getStateInEffectAtSlot_returnStateFromLastBlockWhenSlotsAreEmpty() {
+    storageClient.updateBestBlock(GENESIS_BLOCK_ROOT, UnsignedLong.ONE);
+
+    final BeaconState bestState = dataStructureUtil.randomBeaconState(UnsignedLong.ZERO);
+    final BeaconState newState =
+        bestState.updated(state -> state.getBlock_roots().set(0, GENESIS_BLOCK_ROOT));
+    when(store.getBlockState(GENESIS_BLOCK_ROOT)).thenReturn(newState);
+    when(store.getBlock(GENESIS_BLOCK_ROOT)).thenReturn(GENESIS_BLOCK);
+
+    assertThat(storageClient.getStateInEffectAtSlot(UnsignedLong.ONE)).contains(newState);
+  }
+
+  @Test
+  public void getStateInEffectAtSlot_returnStateFromLastBlockWhenHeadSlotIsEmpty() {
+    storageClient.updateBestBlock(GENESIS_BLOCK_ROOT, UnsignedLong.ZERO);
+
+    final BeaconState bestState = dataStructureUtil.randomBeaconState(UnsignedLong.ZERO);
+    final BeaconState newState =
```

Comment:
I suggest removing lines 199 and 212, and switching the new state declaration lines to:
```suggestion
    final BeaconState newState = dataStructureUtil.randomBeaconState(UnsignedLong.ONE);
```
I don't know why you used the bestState.updated() method, but I removed them and switched to the suggestion I made above locally, and tests were still good. If they don't serve a special purpose, I think they make the tests harder to read.